internal/services/securitycenter/security_center_storage_defender_resource.go

```diff
@@ -120,6 +126,7 @@ func (s StorageDefenderResource) Create() sdk.ResourceFunc {
 							IsEnabled:     pointer.To(plan.MalwareScanningOnUploadEnabled),
 							CapGBPerMonth: pointer.To(plan.MalwareScanningOnUploadCapPerMon),
 						},
+						ScanResultsEventGridTopicResourceId: pointer.To(plan.ScanResultsEventGridTopicId),
```

Comment:
This might only want to be conditionally set in the payload e.g.

```
if plan.ScanResultsEventGridTopicId != "" {
    topicId, err := topics.ParseTopicID(plan.ScanResultsEventGridTopicId)
    if err != nil {
       return err
    }
    input.Properties.MalwareScanning.ScanResultsEventGridTopicResourceId = pointer.To(topicId.ID())
}
```

Reply:
thanks for the hint, I made the suggested change